### PR TITLE
Migrated Key Vault JCA library to use azure-json serialization/deserialization

### PR DESCRIPTION
**Pull request by vcolin7**
This library shades Jackson classes to avoid having many external dependencies, so I've gone ahead and done the same for azure-json. I did copy a number of classes over from azure-core, so maybe there's a better approach for that. Let me know what you think @alzimmermsft.

For more information on this, see: #40614.

### CODE REVIEW

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/JsonConverterUtil.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/JsonConverterUtil.java:35]: We don't really need this in JCA as the set of types being deserialized is known as they're all defined by the package itself (unlike in Core which doesn't know what it's deserializing other than it's a `JsonSerializable` type). Let's remove this and change the deserialization method to `public static Object fromJson(String json, Function<JsonReader, Object> deserializationFunction)`. Also, I think we can change `Object` to `T`.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/AccessToken.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/AccessToken.java:84]: This changed the names of the JSON properties from `access_token` and `expires_in` to `accessToken` and `expiresIn`, this should be changed back to the old names. Same issue is present in the deserialization function.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/CertificateListResult.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/CertificateListResult.java:57]: Mind moving the field declaration to the right spot in the class.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/CertificateListResult.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/CertificateListResult.java:61]: Any reason `value` and `nextLink` aren't being serialized here?

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/KeyProperties.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/KeyProperties.java:90]: Don't need to use `getNullable` for either deserialization calls. If the JSON property `exportable` is `null` then this class has an invalid definition. `getString` always supports returning `null`, so calling `getNullable` is just extra overhead that doesn't accomplish anything.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/SignResult.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/SignResult.java:56]: Any reason `kid` and `value` aren't being serialized?

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/CertificateListResult.java
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/CertificateListResult.java:61]: I copied this over from the Key Vault Certificates library where this type is read-only. I'll add those two here just in case, it's an implementation detail after all.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/SignResult.java
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/model/SignResult.java:56]: Same as the Certificates class. I'll add it for good measure.

**Comment by saragluna** [comment]
Hey @vcolin7, thanks for creating this PR. I am wondering whether we should revert the two changes you made for the AAD Login URL, before we merge in any PR? Seems like you are actively contributing to make the JCA library better, and we also have some features that want to be merged in, and out as regular release. So I think if those two changes won't be out as GA in the near future, maybe we should revert it, and merge it back to main when it's ready to GA.

**Comment by vcolin7** [comment]
> Hey @vcolin7, thanks for creating this PR. I am wondering whether we should revert the two changes you made for the AAD Login URL, before we merge in any PR? Seems like you are actively contributing to make the JCA library better, and we also have some features that want to be merged in, and out as regular release. So I think if those two changes won't be out as GA in the near future, maybe we should revert it, and merge it back to main when it's ready to GA.

I think if you guys want to release a single feature that's already in `main` we should not revert the commits, but instead create a release branch containing only said feature, which is what we do with our other data plane libraries. We have a script that can help you set up the branch and then it would be a matter of cherry-picking the commits that include the feature work. We can talk about this on a separate chat if you need any assistance setting things up.

**Comment by saragluna** [comment]
Thanks @vcolin7. But what I'm talking is we are having a PR https://github.com/Azure/azure-sdk-for-java/pull/41303, we would like to merge into the main branch. But the main branch already has changes only suitable for beta releases. So my question is, should we use the main branch as a place for code ready for GA release, and move the code for beta releases to a feature branch?

**Comment by vcolin7** [comment]
> Thanks @vcolin7. But what I'm talking is we are having a PR #41303, we would like to merge into the main branch. But the main branch already has changes only suitable for beta releases. So my question is, should we use the main branch as a place for code ready for GA release, and move the code for beta releases to a feature branch?

I think we should merge your PR to `main` and then release a new stable version from a separate release branch that does not include the beta changes we are not ready to GA yet. This is because if we release any other beta in the future, it should not be missing features or bug fixes we already released for a "previous" stable version (like this that would be included in a 2.8.2 release). Reverting the commits with the beta work would add unnecessary overhead. We should be getting close to GA'ing the beta work anyway, we're just waiting for customer feedback on it.

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/Base64Url.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/Base64Url.java:1]: Why did we need to copy this from azure-core?

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/Base64Util.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/Base64Util.java:1]: Same with this class

**Review by alzimmermsft — APPROVED** [review]

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/JsonConverterUtil.java
Comment (alzimmermsft) [inline sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/JsonConverterUtil.java:40]: Slight modification to this class to complete remove the need for reflection here.

```java
public static <T extends JsonSerializable<T>> T fromJson(ReadValueCallback<JsonReader, T> deserializationFunction, String json) throws IOException {
    try (JsonReader jsonReader = JsonProviders.createReader(json)) {
        return deserializationFunction(jsonReader);
    }
}
```

then methods call this can change to

```java
AccessToken accessToken = JsonConverterUtil.fromJson(AccessToken::fromJson, jsonString);
```

**Comment by joshfree** [comment]
@vcolin7 what's happening with this PR?  Can we get this wrapped up?

File: sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/Base64Url.java
Comment (vcolin7) [reply sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/Base64Url.java:1]: So we could URL-encode a field in `SignResult`. We can remove it and the accompanying `Base64Util` and just use the code we need.

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**
